blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedIngestionHandler.java

```diff
@@ -99,10 +99,10 @@ public void handle(ChannelHandlerContext ctx, HttpRequest request) {
                         HttpResponseStatus.BAD_REQUEST );
             }
 
-        } catch (JsonParseException ex) {
+        } catch (JsonSyntaxException ex) {
```

Comment:
Ok never mind, while the statement is true that we're using two different json parsing library (Jackson for regular metrics and Gson for aggregated metrics), it's ok to leave this as JsonParseException, since JsonSyntaxException extends it, so when it is throw by gson it still catches it here.

It is however still true that while Jackson are ok with the string '[]', which is valid and we only caught and return 400 because of having no metrics return from the mapping, Gson does not like seeing '[]' at all for non-array expecting metric (i.e. if it's trying to map to a single aggregated metric), and will throw the error "java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BEGIN_ARRAY at line 1 column ".

I'll create a story as a tech debt to choose one or the other to use (Jackson or Gson) and refactor the code so it's more consistent with errors and validations.